manim/mobject/svg/text_mobject.py

```diff
@@ -339,122 +318,213 @@ def text2svg(self):
             if setting.line_num != last_line_num:
                 offset_x = 0
                 last_line_num = setting.line_num
-            context.move_to(START_X + offset_x, START_Y + lsh * setting.line_num)
+            context.move_to(
+                START_X + offset_x, START_Y + line_spacing * setting.line_num
+            )
             context.show_text(text)
             offset_x += context.text_extents(text)[4]
+        surface.finish()
+        return file_name
+
+
+# Following class is just a Little implementation of upcomming feautures. Ignore it for now.
+class TextWithBackground(Text):
+    CONFIG = {
+        "background_color": BLACK,
+    }
+
+    def __init__(self, text, **config):
+        Text.__init__(self, text, **config)
+        # self.text_backgrounds = self.gen_text_backgrounds(text)
+
+    def gen_text_backgrounds(self, text):
+        text_with_visible_chars = text.replace(" ", "").replace("\t", "")
+        text_list = text_with_visible_chars.split("\n")
+        text_backgrounds = VGroup()
+        start_i = 0
+        for line_no in range(text_list.__len__()):
+            rect_counts = len(text_list[line_no])
+            text_backgrounds.add(*self[start_i:rect_counts])
+            start_i += 2 * rect_counts
+        text_backgrounds.set_color(self.background_color)
+
+        return text_backgrounds
+
+    def text2svg1(self):
+        # anti-aliasing
+        size = self.size * 10
+        line_spacing = self.line_spacing * 10
+
+        if self.font == "":
+            if NOT_SETTING_FONT_MSG != "":
+                logger.warning(NOT_SETTING_FONT_MSG)
+        dir_name = consts.TEXT_DIR
+        hash_name = self.text2hash()
+        file_name = os.path.join(dir_name, hash_name) + ".svg"
+        # if os.path.exists(file_name):
+        # return file_name
+
+        surface = cairo.SVGSurface(file_name, 600, 400)
+        context = cairo.Context(surface)
+        context.set_font_size(size)
+        context.move_to(START_X, START_Y)
 
+        settings = self.text2settings()
+        offset_x = 0
+        last_line_num = 0
+        for setting in settings:
+            font = setting.font
+            slant = self.str2slant(setting.slant)
+            weight = self.str2weight(setting.weight)
+            text = self.text[setting.start : setting.end].replace("\n", " ")
+            context.select_font_face(font, slant, weight)
+            if setting.line_num != last_line_num:
+                offset_x = 0
+                last_line_num = setting.line_num
+            tempx = START_X + offset_x
+            tempy = START_Y + line_spacing * setting.line_num
+            char_offset_x = 0
+            char_height = tempy - size / 2 - (line_spacing - size)
+            # context.set_font_matrix(cairo.Matrix(13, 0, 0, 13, 0, 0))
```

Comment:
Why is it commented out?

Reply:
I don't exactly remember but I remember maybe it's from a very old version when I was trying to solve spaces problem in `Text`

Reply:
we can ignore it